src/MatrixEventHandler.ts

```diff
@@ -156,22 +156,41 @@ export class MatrixEventHandler {
                 }
             } else if (event.content.is_direct && bridgeBot.isRemoteUser(event.state_key)) {
                 log.debug("Got request to PM", event.state_key);
-                const {
-                    username,
-                    protocol,
-                } = this.purple.getUsernameFromMxid(event.state_key!, this.config.bridge.userPrefix);
-                log.debug("Mapped username to", username);
-                const remoteData = {
+                let remoteData = {
                     matrixUser: event.sender,
-                    protocol_id: protocol.id,
-                    recipient: username,
                 } as any;
+
+                // this is not always possible (could be a brand new user being reached),
+                // but is preferable to decoding the mxid because of potential information loss
+                // during mxid generation (see GH-268)
+                const remoteUsers = await this.store.getRemoteUsersFromMxId(event.state_key!);
+                if (remoteUsers.length === 1) {
```

Comment:
Should log.error if it's more than one, since it'd indicate something rather unexpected.